awx/ui_next/src/components/ListHeader/ListHeader.jsx

```diff
@@ -38,60 +37,59 @@ class ListHeader extends React.Component {
 
   handleSearch(key, value) {
     const { location, qsConfig } = this.props;
-    let params = parseQueryString(qsConfig, location.search);
-    params = mergeParams(params, { [key]: value });
-    params = replaceParams(params, { page: 1 });
-    this.pushHistoryState(params);
+    const params = parseQueryString(qsConfig, location.search);
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      ...mergeParams(params, { [key]: value }),
+      page: 1,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleReplaceSearch(key, value) {
     const { location, qsConfig } = this.props;
-    const oldParams = parseQueryString(qsConfig, location.search);
-    this.pushHistoryState(replaceParams(oldParams, { [key]: value }));
+    const qs = replaceNamespacedParams(qsConfig, location.search, {
+      [key]: value,
+    });
+    this.pushHistoryState(qs);
   }
 
   handleRemove(key, value) {
     const { location, qsConfig } = this.props;
-    let oldParams = parseQueryString(qsConfig, location.search);
-    if (parseInt(value, 10)) {
```

Comment:
are you sure you don't need to do this conditional anymore?  I would double check that number-based tag removal works correctly.